pandas/_libs/window/aggregations.pyx

```diff
@@ -1592,7 +1592,7 @@ def roll_weighted_var(const float64_t[:] values, const float64_t[:] weights,
 
     with nogil:
 
-        for i in range(win_n):
+        for i in range(min(win_n, n)):
```

Comment:
Will need a unit test for this change (in `test_win_type.py`). You can use the example in your bug report (without the iteration)